R/lint.R

```diff
@@ -34,18 +37,37 @@ NULL
 #' }
 #'
 #' @export
-lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE) {
+lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE, text = NULL) {
 
-  inline_data <- rex::re_matches(filename, rex::rex(newline))
-  if (inline_data) {
-    content <- gsub("\n$", "", filename)
+  if (is.null(text)) {
+    inline_data <- rex::re_matches(filename, rex::rex(newline))
+    if (inline_data) {
+      text <- gsub("\n$", "", filename)
+      filename <- NULL
+    }
+  } else {
+    inline_data <- TRUE
+    if (length(text) > 1) {
+      text <- paste0(text, collapse = "\n")
```

Comment:
nit: i usually prefer `paste` to `paste0` when we're just aggregating. I don't think there's any technical difference, but I feel like I have to add "`sep=""`" mentally and then remove it when I see there's only one argument, if that makes sense?

Reply:
Yes it is more readable and clear about the intended behavior.